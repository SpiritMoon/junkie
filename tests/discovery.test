#!/bin/sh
# Test that:
# - tlsv1.pcap: detect most https packets as TLSv1
# - single-rtp.pcap: a single RTP packet (without signalisation) is passed to RTP parser

NAME="Discovery"
# Check we can discover these
PCAPS="pcap/https/tlsv1.pcap pcap/voip/single-rtp.pcap pcap/misc/telnet.pcap pcap/misc/samba.pcap pcap/misc/irc.pcap pcap/misc/rdp.pcap pcap/dns/dns.pcap"
# Check we also find and follow these protos on non-standard ports
PCAPS="$PCAPS pcap/http/on-alt-port.pcap pcap/ftp/on-alt-port.pcap pcap/sip/on-alt-port.pcap"
# This one match the signature but the proper parser should fail
PCAPS="$PCAPS pcap/http/not-http.pcap"
# Check SMTP signature
PCAPS="$PCAPS pcap/email/smtp.pcap"
# Check sql
PCAPS="$PCAPS pcap/oracle/one_row_response.pcap pcap/postgres/connect.pcap pcap/postgres/long_select.pcap"
# Check nfs
PCAPS="$PCAPS pcap/nfs/nfs.pcap"

filter() {
	grep -wE '^(PIPI|TLS|RTP|HTTP|FTP|SIP|CIFS|DNS|Netbios|TNS|PostgreSQL|RPC)'
}
CMDLINE='(use-modules (junkie signatures))(tcp-del-port "tns" 1521)(tcp-del-port "PostgreSQL" 5432)(tcp-del-port "tls" 443)(tcp-del-port "Netbios" 445)(udp-del-port "DNS" 53)(udp-del-port "RPC" 2049)(tcp-del-port "RPC" 2049)'
OUTPUT="discovery.out"

. "$srcdir/dump_test"
