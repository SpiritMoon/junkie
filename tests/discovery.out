-- pcap/https/tlsv1.pcap --
TLS: head_len=167, payload=0, version=3.1, content-type=handshake
TLS: head_len=70, payload=0, version=3.1, content-type=handshake, cipher_suite=Unknown cipher suite 0xc011, compression_algo=none
TLS: head_len=1630, payload=0, version=3.1, content-type=handshake, CN=www.google.com
TLS: head_len=75, payload=0, version=3.1, content-type=handshake
TLS: head_len=179, payload=0, version=3.1, content-type=handshake
TLS: head_len=1138, payload=0, version=3.1, content-type=data
TLS: head_len=936, payload=0, version=3.1, content-type=data
-- pcap/voip/single-rtp.pcap --
RTP: head_len=12, payload=160, payload_type=PCMU, SSRC=1438910670, seqnum=23777, timestamp=2240
-- pcap/misc/telnet.pcap --
PIPI: head_len=0, payload=33, protocol=17(Telnet), trust=low
PIPI: head_len=0, payload=12, protocol=17(Telnet), trust=low
PIPI: head_len=0, payload=45, protocol=17(Telnet), trust=low
-- pcap/misc/samba.pcap --
Netbios: head_len=4, payload=74
CIFS: head_len=32, payload=42, command=0x32, status=0x00000000
Netbios: head_len=4, payload=160
CIFS: head_len=32, payload=128, command=0x32, status=0x00000000
Netbios: head_len=4, payload=84
CIFS: head_len=32, payload=52, command=0x32, status=0x00000000
-- pcap/misc/irc.pcap --
PIPI: head_len=0, payload=12, protocol=11(IRC), trust=low
-- pcap/misc/rdp.pcap --
PIPI: head_len=0, payload=418, protocol=22(RDP), trust=high
-- pcap/dns/dns.pcap --
DNS: head_len=32, payload=0, QUERY, tx_id=3804, err_code=0, request_type=A, dns_class=IN, name=www.google.com
DNS: head_len=32, payload=0, QUERY, tx_id=51952, err_code=0, request_type=AAAA, dns_class=IN, name=www.google.com
DNS: head_len=284, payload=0, ANSWER, tx_id=3804, err_code=0, request_type=A, dns_class=IN, name=www.google.com
-- pcap/http/on-alt-port.pcap --
HTTP: head_len=95, payload=0, method=GET, code=unset, content_length=unset, transfert_encoding=unset, mime_type=unset, host=192.168.1.1:8080, user_agent=RobiNet, referrer=unset, server=unset, orig_ip=unset, url=/, pkts=1
HTTP: head_len=67, payload=0, method=unset, code=200, content_length=1011, transfert_encoding=unset, mime_type=text/plain, host=unset, user_agent=unset, referrer=unset, server=unset, orig_ip=unset, url=unset, pkts=1
-- pcap/ftp/on-alt-port.pcap --
FTP: head_len=0, payload=14
FTP: head_len=0, payload=26
FTP: head_len=0, payload=7
FTP: head_len=0, payload=37
FTP: head_len=0, payload=8
FTP: head_len=0, payload=30
FTP: head_len=0, payload=27
FTP: head_len=0, payload=51
FTP: head_len=0, payload=11
FTP: head_len=0, payload=61
FTP: head_len=0, payload=24
-- pcap/sip/on-alt-port.pcap --
SIP: head_len=584, payload=183, cmd=INVITE, cseq=50328871, via=UDP 212.27.52.5:5060, code=unset, mime_type=application/sdp, content-length=183, call-id=20812-KJ-030ff678-1ebf056e3@freephonie.net, from="Sebastien Douche" <sip:0178090721@freephonie.net;user=phone>;tag=20812-TF-030ff679-135e3e8b5, to=<sip:0950652801@172.17.20.241;user=phone>
SIP: head_len=453, payload=0, cmd=unset, cseq=50328871, via=UDP 212.27.52.5:5060, code=100, mime_type=unset, content-length=0, call-id=20812-KJ-030ff678-1ebf056e3@freephonie.net, from="Sebastien Douche" <sip:0178090721@freephonie.net;user=phone>;tag=20812-TF-030ff679-135e3e8b5, to=<sip:0950652801@172.17.20.241;user=phone>
SIP: head_len=648, payload=0, cmd=unset, cseq=50328871, via=UDP 212.27.52.5:5060, code=180, mime_type=unset, content-length=0, call-id=20812-KJ-030ff678-1ebf056e3@freephonie.net, from="Sebastien Douche" <sip:0178090721@freephonie.net;user=phone>;tag=20812-TF-030ff679-135e3e8b5, to=<sip:0950652801@172.17.20.241;user=phone>;tag=2Xg1tg75j8gDH
SIP: head_len=399, payload=0, cmd=CANCEL, cseq=50328871, via=UDP 212.27.52.5:5060, code=unset, mime_type=unset, content-length=0, call-id=20812-KJ-030ff678-1ebf056e3@freephonie.net, from="Sebastien Douche" <sip:0178090721@freephonie.net;user=phone>;tag=20812-TF-030ff679-135e3e8b5, to=<sip:0950652801@172.17.20.241;user=phone>
SIP: head_len=352, payload=0, cmd=unset, cseq=50328871, via=UDP 212.27.52.5:5060, code=200, mime_type=unset, content-length=0, call-id=20812-KJ-030ff678-1ebf056e3@freephonie.net, from="Sebastien Douche" <sip:0178090721@freephonie.net;user=phone>;tag=20812-TF-030ff679-135e3e8b5, to=<sip:0950652801@172.17.20.241;user=phone>;tag=2Xg1tg75j8gDH
SIP: head_len=521, payload=0, cmd=unset, cseq=50328871, via=UDP 212.27.52.5:5060, code=487, mime_type=unset, content-length=0, call-id=20812-KJ-030ff678-1ebf056e3@freephonie.net, from="Sebastien Douche" <sip:0178090721@freephonie.net;user=phone>;tag=20812-TF-030ff679-135e3e8b5, to=<sip:0950652801@172.17.20.241;user=phone>;tag=2Xg1tg75j8gDH
-- pcap/http/not-http.pcap --
-- pcap/email/smtp.pcap --
PIPI: head_len=0, payload=34, protocol=24(SMTP), trust=medium
TLS: head_len=117, payload=0, version=3.0, content-type=handshake
TLS: head_len=86, payload=0, version=3.1, content-type=handshake, cipher_suite=DHE_RSA_WITH_AES_128_CBC_SHA, compression_algo=none
TLS: head_len=139, payload=0, version=3.1, content-type=handshake
TLS: head_len=6, payload=0, version=3.1, content-type=change_cipher_spec
TLS: head_len=197, payload=0, version=3.1, content-type=handshake
TLS: head_len=6, payload=0, version=3.1, content-type=change_cipher_spec
TLS: head_len=245, payload=0, version=3.1, content-type=data
TLS: head_len=165, payload=0, version=3.1, content-type=data
TLS: head_len=149, payload=0, version=3.1, content-type=data
TLS: head_len=101, payload=0, version=3.1, content-type=data
TLS: head_len=661, payload=0, version=3.1, content-type=data
TLS: head_len=69, payload=0, version=3.1, content-type=data
TLS: head_len=53, payload=0, version=3.1, content-type=data
TLS: head_len=149, payload=0, version=3.1, content-type=alert
TLS: head_len=53, payload=0, version=3.1, content-type=data
TLS: head_len=37, payload=0, version=3.1, content-type=alert
-- pcap/oracle/one_row_response.pcap --
TNS: head_len=165, payload=0, Clt->Srv, version=3.8, startup, No SSL, user=
TNS: head_len=8, payload=0, Srv->Clt, unknown
TNS: head_len=165, payload=0, Clt->Srv, version=3.8, startup, No SSL, user=
TNS: head_len=24, payload=0, Srv->Clt, version=3.8, startup, No SSL, Status=Completed
TNS: head_len=152, payload=0, Clt->Srv, unknown
TNS: head_len=127, payload=0, Srv->Clt, unknown
TNS: head_len=33, payload=0, Clt->Srv, unknown
TNS: head_len=237, payload=0, Srv->Clt, unknown
TNS: head_len=779, payload=0, Clt->Srv, unknown
TNS: head_len=834, payload=0, Srv->Clt, unknown
TNS: head_len=19, payload=0, Clt->Srv, unknown
TNS: head_len=71, payload=0, Srv->Clt, unknown
TNS: head_len=159, payload=0, Clt->Srv, unknown
TNS: head_len=73, payload=0, Srv->Clt, unknown
TNS: head_len=568, payload=0, Clt->Srv, unknown
TNS: head_len=1169, payload=0, Srv->Clt, unknown
TNS: head_len=76, payload=0, Clt->Srv, query, query='SELECT * FROM "Toto"'
TNS: head_len=443, payload=0, Srv->Clt, query, nb_rows=0, nb_fields=2
TNS: head_len=17, payload=0, Clt->Srv, query
TNS: head_len=90, payload=0, Srv->Clt, query, nb_rows=1, nb_fields=-1, Status=Completed, ErrorCode=ORA-01403, ErrorMessage=no data found
-- pcap/postgres/connect.pcap --
PostgreSQL: head_len=34, payload=0, Clt->Srv, version=3.0, startup, No SSL, user=rixed, dbname=nova
PostgreSQL: head_len=13, payload=0, Srv->Clt, startup, No SSL
PostgreSQL: head_len=41, payload=0, Clt->Srv, startup, No SSL, passwd=md5c0b295f0cd4b08167fbf02c5d0c1345f
PostgreSQL: head_len=303, payload=0, Srv->Clt, startup, No SSL, Status=Completed
PostgreSQL: head_len=5, payload=0, Clt->Srv, exit, Status=Completed
-- pcap/postgres/long_select.pcap --
PostgreSQL: head_len=8, payload=0, Clt->Srv, startup, SSL requested
PostgreSQL: head_len=1, payload=0, Srv->Clt, startup, SSL refused
PostgreSQL: head_len=33, payload=0, Clt->Srv, version=3.0, startup, No SSL, user=sora, dbname=sora
PostgreSQL: head_len=324, payload=0, Srv->Clt, startup, No SSL, Status=Completed
PostgreSQL: head_len=285, payload=0, Clt->Srv, query, query='SELECT * FROM                                                                                                                                                                                                                                                                octonions;'
PostgreSQL: head_len=71, payload=0, Srv->Clt, query, nb_rows=0, nb_fields=2, Status=Completed
PostgreSQL: head_len=5, payload=0, Clt->Srv, exit, Status=Completed
